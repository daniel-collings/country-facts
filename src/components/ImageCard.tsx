import React from 'react'

interface IImageCardProps {
  src: string
  alt: string
}

export default function ImageCard({ src, alt }: IImageCardProps) {
  return (
    <div className="bg-base-300 rounded-xl p-4">
      <img src={src} alt={alt} className="w-32 h-auto" />
    </div>
  )
}
