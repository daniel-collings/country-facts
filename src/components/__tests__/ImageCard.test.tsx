import { describe, expect, test } from 'vitest'
import { render } from '@/utils/testUtils.tsx'
import ImageCard from '@/components/ImageCard.tsx'

describe('ImageCard Component', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<ImageCard src={'/vite.svg'} alt="vite" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
