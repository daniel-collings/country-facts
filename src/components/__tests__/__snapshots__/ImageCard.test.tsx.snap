// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageCard Component > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="h-full min-h-screen bg-base-100 p-4 sm:p-8"
    data-theme="light"
  >
    <main
      class="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8"
    >
      <div
        class="bg-base-300 rounded-xl p-4"
      >
        <img
          alt="vite"
          class="w-32 h-auto"
          src="/vite.svg"
        />
      </div>
    </main>
  </div>
</DocumentFragment>
`;
